pkg/prometheus/statefulset.go

```diff
@@ -211,10 +210,6 @@ func makeStatefulSet(
 		statefulset.Spec.VolumeClaimTemplates = append(statefulset.Spec.VolumeClaimTemplates, pvcTemplate)
 	}
 
-	// Updates to statefulset spec for fields other than 'replicas',
-	// 'template', and 'updateStrategy' are forbidden.
-	statefulset.Spec.PodManagementPolicy = previousPodManagementPolicy
```

Comment:
I am a bit confused. Given that Kubernetes defaults to `orderedready` and the fact that we would like `parallel`, don't we need to set `parallel` here?

Reply:
Ah, nevermind, this is set in https://github.com/coreos/prometheus-operator/blob/7d73503f0f0be5565b0e017fea04a6c753c24432/pkg/prometheus/statefulset.go#L753